### PR TITLE
simulatioion: Declare dependency on asdf-cxx

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

File: var/spack/repos/builtin/packages/simulationio/package.py [hunk 1]
```diff
@@ -23,6 +23,8 @@ class Simulationio(CMakePackage):
     variant('pic', default=True,
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/simulationio/package.py:25]: Can you add descriptions to these variants? And rename develop to master?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/simulationio/package.py:25]: Ping @eschnett

File: var/spack/repos/builtin/packages/simulationio/package.py [hunk 2]
```diff
@@ -14,15 +14,17 @@ class Simulationio(CMakePackage):
     url      = "https://github.com/eschnett/SimulationIO/archive/version/9.0.1.tar.gz"
     git      = "https://github.com/eschnett/SimulationIO.git"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/simulationio/package.py:15]: Want to add yourself as a maintainer?
```python
maintainers = ['eschnett']
```

File: var/spack/repos/builtin/packages/simulationio/package.py (hunk 1 above)
Comment (eschnett) [reply var/spack/repos/builtin/packages/simulationio/package.py:25]: Done.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**